File: JavaRobot/src/org/usfirst/frc/team195/robot/Robot.java

package org.usfirst.frc.team195.robot;

import edu.wpi.first.wpilibj.SampleRobot;


public class Robot extends SampleRobot {

	public Robot() {
	}

	@Override
	public void robotInit() {
		myDrive = new DriveBase();
		myIntake = new Intake();
		
	}
	
	@Override
	public void autonomous() {

	}

	@Override
	public void operatorControl() {
		while (isOperatorControl() && isEnabled()) {
			myDrive.run();
			myIntake.run();
		}
	}


	@Override
	public void test() {
	}
	
	private DriveBase myDrive;
	private Intake myIntake;
}

File: ParadeRobots/KnightHawk_Final/src/Robot.cpp
#include "Robot.h"

void Robot::RobotInit() {
	robotControllers = new Controllers();
	ballIntake = new Intake(robotControllers);
	ballShooter = new Shooter(robotControllers);

	initComplete = false;
}

void Robot::OperatorControl() {
	if(IsEnabled()&&IsOperatorControl())
	{
		//First run code
		if (!initComplete)
		{
			if(!ballShooter->isHomed()) {
				ballShooter->home();
			}
			initComplete = true;
		}

		ballIntake->start();
		ballShooter->start();
	}
	while (IsEnabled()&&IsOperatorControl()) {
		//std::cout << robotControllers->getWinchSwitch()->Get() << std::endl;
		std::this_thread::sleep_for(std::chrono::milliseconds(100));
	}
}

START_ROBOT_CLASS(Robot)

